.gitignore

```diff
@@ -73,3 +73,7 @@ out/
 target/
 /.sonarlint/
 /report.json
+
+# ANTLR grammar
+!/server/grammar/*.g4
+/server/grammar/
```

Comment:
Always touchy.

* I’d put the inclusion after the exclusion.

> It is not possible to re-include a file if a parent directory of that file is excluded.

(https://git-scm.com/docs/gitignore#_pattern_format)

* I’d use a `*` in the exclusion.

> otherwise you're ignoring the directory itself (so git won't look inside) instead of the files within the directory (which allows for the exclusion).

(https://stackoverflow.com/a/3001967)

My understanding of these things is not perfect, though.